staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -169,8 +169,7 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attrib
 			switch err := err.(type) {
 			case *webhookutil.ErrCallingWebhook:
 				if ctx.Err() == context.Canceled {
```

Comment:
did you look into checking `err` instead of ctx.Err?

Reply:
Yes I checked and added the comments in the old PR https://github.com/kubernetes/kubernetes/pull/119341

Reply:
The err that is from callAttrMutatingHook method [[ref](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go#L165)] based on the provided ctx . The err.Reason returns customized error response that picks up the ctx error so I don’t think anything wrong by checking this way ctx.Err() == context.Canceled as it is similar to checking the err obj.

Also The err.Reason returns
%!(EXTRA string=failed to call webhook: Post "https://127.0.0.1:21375/validate?timeout=2s": context canceled) which is a customized error and may break if someone modifies the error message so I think its better to go with checking ctx.Err().